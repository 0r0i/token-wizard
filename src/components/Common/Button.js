import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
