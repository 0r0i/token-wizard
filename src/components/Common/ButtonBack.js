import React, { Component } from 'react'

export class ButtonBack extends Component {
  /**
   * Render method for ButtonBack component
   * @returns {*}
   */
  render() {
    const { onClick } = this.props

    return (
      <button onClick={onClick} type="button" className={`sw-ButtonBack`}>
        Back
      </button>
    )
  }
}
