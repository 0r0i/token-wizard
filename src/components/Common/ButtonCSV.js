import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
