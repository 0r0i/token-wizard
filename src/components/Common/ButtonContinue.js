import React from 'react'

export const ButtonContinue = ({ disabled = false, type, onClick, extraClassName, buttonText = 'Continue' }) => (
  <button
    className={`sw-ButtonContinue ${extraClassName ? extraClassName : ''}`}
    disabled={disabled}
    onClick={onClick}
    type={type}
  >
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
