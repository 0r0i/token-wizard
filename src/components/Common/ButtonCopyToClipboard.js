import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
