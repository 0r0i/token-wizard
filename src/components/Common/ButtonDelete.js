import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
