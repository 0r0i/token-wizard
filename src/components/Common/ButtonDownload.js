import React, { Component } from 'react'

export class ButtonDownload extends Component {
  render() {
    const { disabled, onClick, extraClass } = this.props
    return (
      <button
        className={`sw-ButtonDownload ${extraClass ? extraClass : ''}`}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        Download
      </button>
    )
  }
}
