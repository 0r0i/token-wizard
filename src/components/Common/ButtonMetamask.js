import React from 'react'

export const ButtonMetamask = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`btn-ButtonMetamask ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    <span className="btn-ButtonMetamask_Icon" />
    <span className="btn-ButtonMetamask_TextContainer">
      <span className="btn-ButtonMetamask_TopText">Download</span>
      <span className="btn-ButtonMetamask_BottomText">Metamask</span>
    </span>
  </button>
)
