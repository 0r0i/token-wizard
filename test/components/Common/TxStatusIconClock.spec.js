import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, shallow } from 'enzyme'
import { TxStatusIconClock } from '../../../src/components/Common/TxStatusIconClock'

configure({ adapter: new Adapter() })

describe(`TxStatusIconClock`, () => {
  it(`should be a <svg> tag`, () => {
    const wrapper = shallow(<TxStatusIconClock />)

    expect(wrapper.type()).toBe('svg')
  })

  it(`should render TxStatusIconClock component`, () => {
    // Given
    const component = render.create(<TxStatusIconClock />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
