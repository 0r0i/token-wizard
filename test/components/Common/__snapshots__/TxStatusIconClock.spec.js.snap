// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxStatusIconClock should render TxStatusIconClock component 1`] = `
<svg
  className="tx-TxStatusIconClock"
  height="19"
  viewBox="0 0 19 19"
  width="19"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="#404040"
    fillRule="nonzero"
    opacity=".503"
  >
    <path
      d="M9.419 0C4.225 0 0 4.225 0 9.419c0 5.193 4.225 9.418 9.419 9.418 5.193 0 9.418-4.225 9.418-9.418C18.837 4.225 14.612 0 9.42 0zm0 16.833c-4.089 0-7.415-3.326-7.415-7.414 0-4.089 3.326-7.415 7.415-7.415 4.088 0 7.414 3.326 7.414 7.415 0 4.088-3.326 7.414-7.414 7.414z"
    />
    <path
      d="M14.327 9.128h-4.233v-5.09a.776.776 0 0 0-1.55 0v5.865c0 .429.347.776.775.776h5.008a.776.776 0 0 0 0-1.551z"
    />
  </g>
</svg>
`;
